### PR TITLE
#4 As a registered user, I want to like a post

### LINKED ISSUE #4: As a registered user, I want to like a post

### LINKED ISSUE #14: Initialize nistagram-posts backend

### LINKED ISSUE #15: Implement backend methods for "Like a post" feature
User story: #4

### LINKED ISSUE #16: Create frontend components for "Like a post" feature
User story: #4

### PR DESCRIPTION
**Pull request by MasterRadule**
Closes #14, closes #15, closes #16

Resolves #4

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,12 @@
+module github.com/Nistagram-Organization/nistagram-posts
+
+go 1.16
+
+require (
+	github.com/Nistagram-Organization/agent-shared v0.0.0-20210622134726-e03ddfa46b30
```
Comment (draganculibrk9) [inline go.mod]: Remove dependency

File: src/datasources/mysql/agent_application_db.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package mysql
```
Comment (draganculibrk9) [inline src/datasources/mysql/agent_application_db.go]: Rename file, this is not agent application

File: src/repositories/like/like_repository.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+func (l *likesRepository) GetByUserAndPost(userId uint, postId uint) (*like.Like, rest_error.RestErr) {
+	likeEntity := like.Like{
+		UserID: userId,
+		PostID: postId,
+	}
+	if err := l.db.Where("user_id = ? AND post_id = ?", userId, postId).First(&likeEntity).Error; err != nil {
+		fmt.Sprintln(err)
+		return nil, rest_error.NewNotFoundError(fmt.Sprintf("Post has not been liked by user"))
```
Comment (draganculibrk9) [inline src/repositories/like/like_repository.go:33]: No need for Sprintf

File: src/repositories/like/like_repository.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+func (l *likesRepository) Create(like *like.Like) rest_error.RestErr {
+	if err := l.db.Create(like).Error; err != nil {
+		return rest_error.NewInternalServerError("Error when trying to like a post", err)
+	}
+	return nil
+}
+
+func (l *likesRepository) Delete(like *like.Like) rest_error.RestErr {
+	if err :=l.db.Where("user_id = ? AND post_id = ?", like.UserID, like.PostID).Delete(like).Error; err != nil {
```
Comment (draganculibrk9) [inline src/repositories/like/like_repository.go:46]: Where statement seems unnecessary

**Review by draganculibrk9 — CHANGES_REQUESTED** [review]

File: src/repositories/like/like_repository.go (hunk 4 above)
Comment (MasterRadule) [reply src/repositories/like/like_repository.go:46]: It is necessary otherwise it shows "where condition missing" error.

**Review by draganculibrk9 — APPROVED** [review]

**Pull request merged**